src/librustdoc/html/markdown.rs

```diff
@@ -451,9 +479,63 @@ impl<'a> fmt::String for MarkdownWithToc<'a> {
     }
 }
 
+pub fn plain_summary_line(md: &str) -> String {
+    extern "C" fn link(_ob: *mut hoedown_buffer,
+                       _link: *const hoedown_buffer,
+                       _title: *const hoedown_buffer,
+                       content: *const hoedown_buffer,
+                       opaque: *mut libc::c_void) -> libc::c_int
+    {
+        unsafe {
+            if !content.is_null() && (*content).size > 0 {
+                // FIXME(liigo): I don't know why the parameter `_ob` is
+                // not the value passed in by `hoedown_document_render`.
+                // I have to manually pass in `ob` through `opaque` currently.
```

Comment:
I think this is ok to remove, I believe that hoedown generally passes around intermediate buffers as the first argument as it's frequently rendering into other elements, so in this case you're just redirecting all output to one known location instead of intermediate locations.